docs/src/userguide/external_C_code.rst

```diff
@@ -339,42 +339,57 @@ Public Declarations
 ---------------------
 
 You can make C types, variables and functions defined in a Cython module
-accessible to C code that is linked with the module, by declaring them with
-the public keyword::
+accessible to C code that is linked together with the Cython-generated C file,
```

Comment:
To be clear, "the Cython-generated C file" is not what's linked of course.  Though I'm struggling to come up with more concise wording than "the object file compiled from the Cython-generated C file"